### PR TITLE
Gunlights are back (probably)

### LINKED ISSUE #17474: Seclite isn't shown on pKA after being affixed to pKA
### Severity

Low

### Short Description

a pKA with a seclite attached will not show the seclite, but will still give option to turn seclite on/off

### Reproduction

1.) attach seclite to pka
2.) wow
3.) seclite png not on pKA anymore

### Round ID

any

### Test Merges

_No response_

### BYOND Username

_No response_

### Was this done on a locally hosted or non-Yogstation hosted server?

None

### Additional information

probably because of the recent seclite new sec gun fix

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request

Not gonna lie, I have no idea if this works right now, I am just guessing, but it seems like this was the issue.
a certain SOMEONE, not naming names (ling) might have removed our gunlight overlay sytem when updating our lighting to be not shit in #17271
Also closes #17474

thanks ynot for pointing out where the issue came from
Anyways, hopefully this fixes that.

# Changelog

:cl:  
bugfix: Gunlights fixed
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
ill be able to test this when im not at my real job.

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -609,6 +609,23 @@
 	update_gunlight()
 
 /obj/item/gun/proc/update_gunlight()
+	if(gun_light)
+		if(gun_light.on)
```
Comment (ThatLing) [inline code/modules/projectiles/gun.dm]:
```suggestion
		set_light_on(gun_light.on)
```

Comment (ThatLing) [reply code/modules/projectiles/gun.dm]: Might have to add ``light_system = MOVABLE_LIGHT`` in ``/obj/item/gun``, as well as ``light_range`` and ``light_power``

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -609,6 +609,23 @@
...
+		cut_overlay(flashlight_overlay, TRUE)
+		var/state = "flight[gun_light.on? "_on":""]"	//Generic state.
+		if(gun_light.icon_state in icon_states('icons/obj/guns/flashlights.dmi'))	//Snowflake state?
+			state = gun_light.icon_state
+		flashlight_overlay = mutable_appearance('icons/obj/guns/flashlights.dmi', state)
+		flashlight_overlay.pixel_x = flight_x_offset
+		flashlight_overlay.pixel_y = flight_y_offset
+		add_overlay(flashlight_overlay, TRUE)
+	else
+		set_light(0)
```
Comment (ThatLing) [inline code/modules/projectiles/gun.dm]: set_light_on(FALSE)

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -609,6 +609,23 @@
 	update_gunlight()
 
 /obj/item/gun/proc/update_gunlight()
+	if(gun_light)
+		if(gun_light.on)
+			set_light(gun_light.light_range)
+		else
+			set_light(0)
+		cut_overlay(flashlight_overlay, TRUE)
```
Comment (ThatLing) [inline code/modules/projectiles/gun.dm:613]: Delete all this overlay crap

**Review by ThatLing — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun.dm (hunk 3 above)
Comment (ThatLing) [reply code/modules/projectiles/gun.dm:613]: I have no clue what this overlay is for, if it's for lighting you should remove it. Otherwise keep it

**Comment by MajManatee** [comment]
I think the code ALSO handled the light movement itself, and I just forgot to remove that bit since its handled elsewhere

**Review by ThatLing — CHANGES_REQUESTED** [review]
You removed the ``set_light_on`` and didn't add the vars

**Comment by MajManatee** [comment]
I reckon thats all thats needed

**Comment by MajManatee** [comment]
The light is handled elsewhere, all this is doing is handling the item overlay, so i removed the set_light_on stuff completley

**Review by ThatLing — APPROVED** [review]

**Comment by Moltijoe** [comment]
don't think knife is shown on the PKA either, but that can be a different PR

**Pull request merged**